### ISSUE TITLE
[KeyVault] Unable to create self-signed certificate with Azure CLI 2.0 with PowerShell

### ISSUE THREAD

**Issue by MaximRouiller** [Azure/azure-cli#4626]
### Description

When trying to create a self-signed certificate, it generates a JSON parsing error.

```bash
az keyvault certificate create --vault-name my-keyvault -n cert1 -p "$(az keyvault certificate get-default-policy)"
```

I also tried to put the actual JSON content instead of the command interpolation. I also tried to escape the actual JSON content, etc. 

Nothing seems to work. 

```none
Expecting property name enclosed in double quotes: line 1 column 5 (char 4)
```

What I'm expecting is to be able to either pass the filename in a parameter so that it can be read and deserialized properly or pass the content of the file in a reasonably formatted fashion.

---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: MSI

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here:
azure-cli (2.0.18)

acr (2.0.12)
acs (2.0.16)
appservice (0.1.17)
backup (1.0.1)
batch (3.1.4)
billing (0.1.5)
cdn (0.0.9)
cloud (2.0.8)
cognitiveservices (0.1.8)
command-modules-nspkg (2.0.1)
component (2.0.7)
configure (2.0.11)
consumption (0.1.5)
container (0.1.11)
core (2.0.17)
cosmosdb (0.1.13)
dla (0.0.12)
dls (0.0.14)
eventgrid (0.1.4)
extension (0.0.3)
feedback (2.0.6)
find (0.2.7)
interactive (0.3.10)
iot (0.1.12)
keyvault (2.0.11)
lab (0.0.11)
monitor (0.0.10)
network (2.0.15)
nspkg (3.0.1)
profile (2.0.13)
rdbms (0.0.7)
redis (0.2.9)
resource (2.0.15)
role (2.0.12)
servicefabric (0.0.4)
sql (2.0.12)
storage (2.0.16)
vm (2.0.15)

**OS Version:** What OS and version are you using?  
Answer here: Windows 10 version 1703 (OS Build 15063.632)

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: PowerShell

**Comment by MaximRouiller** [comment]
@cecilphillip

**Comment by tjprescott** [comment]
Few points:
1. the example is written for bash, so that syntax won't work for PS.
2. If you save the JSON output of `get-default-policy` to a file, then you would reference that file by: `--policy @yourfile.json`
3. If that doesn't work, try either bash or CMD, as PS is not currently a supported shell.

**Comment by MaximRouiller** [comment]
1. I get that. The documentation should be clearer about that. 

2. Doesn't work as-is if you use @yourfile.json because @ is a PS special character.

So I tried `@myfile.json and it gives me:
> Expecting value: line 1 column 1 (char 0)

3. Same error in CMD. Only works in Bash. I believe that it's a bug. At a bare minimum, this scenario should be supported under Windows without beta features.

**Comment by tjprescott** [comment]
I just confirmed this works in CMD:
```
az keyvault certificate get-default-policy > myfile.json
az keyvault certificate create --vault-name {vault-name} -n cert1 -p @myfile.json
```

> So I tried `@myfile.json and it gives me:
> Expecting value: line 1 column 1 (char 0)

Yes, I got that error too when I put a backtick (`) in front of the @ sign. If you get rid of that it should work fine.

**Comment by tjprescott** [comment]
After discussion off-line, this was occurring because PS saves the output of `get-default-policy` in a different encoding from that of bash and CMD.  Since PS is not a supported shell, I will close this issue.

**Comment by MaximRouiller** [comment]
Issue:

Doing `az keyvault certificate get-default-policy > defaultpolicy.json` would generate a file UTF-16LE encoded under PowerShell.

Although, azure-cli is supposed to handle it... my bet is that it fails to detect UTF-16LE and uses UTF-16 instead which transforms it to a deserialization error instead of an encoding error.

Workaround for those walking this path in the future:

```powershell
az keyvault certificate get-default-policy | Out-File -Encoding utf8 defaultpolicy.json
```

This forces the encoding to UTF-8 and the `az` command can now be used properly in PowerShell like so by escaping the `@`.

```powershell
az keyvault certificate create --vault-name my-keyvault -n cert --policy `@defaultpolicy.json
```

**Comment by tjprescott** [comment]
Fixed by #4628.
